scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -895,6 +895,12 @@ def apply_filters(self, **kwargs):
                     if ts.arch_exclude and plat.arch in ts.arch_exclude:
                         instance.add_filter("In test case arch exclude", Filters.TESTSUITE)
 
+                    if ts.vendor_allow and plat.vendor not in ts.vendor_allow:
```

Comment:
when `plat.vendor` is empty (not specified) the platform will be excluded - It looks correct, but better to document it for the new test attribute.

Reply:
pedantic, but sure, why not.